oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java

```diff
@@ -211,9 +214,14 @@ public TraceBrief queryBasicTraces(long startSecondTB,
             });
             if (StringUtil.isNotEmpty(serviceId)) {
                 final IDManager.ServiceID.ServiceIDDefinition serviceIDDefinition = IDManager.ServiceID.analysisId(
-                    serviceId);
+                        serviceId);
                 swSpan.setServiceCode(serviceIDDefinition.getName());
             }
+            if (StringUtil.isNotEmpty(serviceInstanceId)) {
```

Comment:
This seems `serviceInstanceName` still could be NULL.

Reply:
If you are 100% saying this is not null, why need `if (StringUtil.isNotEmpty(serviceInstanceId))`

Reply:
```
type Span {
    traceId: ID!
    segmentId: ID!
    spanId: Int!
    parentSpanId: Int!
    refs: [Ref!]!
    serviceCode: String!
    serviceInstanceName: ID!
    startTime: Long!
    endTime: Long!
    endpointName: String
    # There are three span types: Local, Entry and Exit
    type: String!
    # Peer network id, e.g. host+port, ip+port
    peer: String
    component: String
    isError: Boolean
    # There are 5 layers: Unknown, Database, RPCFramework, Http, MQ and Cache
    layer: String
    tags: [KeyValue!]!
    logs: [LogEntity!]!
}
```
The processing logic here is just to be consistent with the serviceId

Reply:
I am saying from the code perspective. What do you want to say to refer the GraphQL?

Reply:
What I mean is that GraphQL also defines that the serviceCode cannot be empty, but it also makes null processing when processing the serviceId. I am not very clear about the reason, so I refer to the processing logic of the serviceId